src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java

```diff
@@ -158,10 +159,27 @@ public void generateNewPasswordForUser(String email) throws MessagingException,
         User user = userRepository.findOneByEmail(email).orElseThrow(
                 () -> new UserDoesNotExistException("User with this email does not exist"));
 
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
         user.setPassword(PasswordGenerator.generatePassword());
         String content = emailContent.createContent(user.getPassword(), "classpath:email/forgotPasswordEmail.html");
         emailSender.sendEmail(user, "Forgot Password", content);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+
+        userRepository.save(user);
```

Comment:
Save the user first and then send email. And you should to refactor little bit sendEmail method to receive not a user, but email or emails. That way is more clear what the code does